### PR TITLE
#66: Implemented iCarol requests scheduler

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: src/main/java/org/benetech/servicenet/adapter/eden/model/BaseData.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.benetech.servicenet.adapter.eden.model;
+
+import lombok.Data;
+
+@Data
+public class BaseData {
+
+    private String id;
+
+    private String databaseID;
```
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/BaseData.java]: it has to be named like this or I'll have to add annotation SerializedName("databaseID") above the field, which seems to be a worse solution

File: src/main/java/org/benetech/servicenet/adapter/eden/model/Weekday.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package org.benetech.servicenet.adapter.eden.model;
+
+public enum Weekday {
+    MON(0),
+    TUE(1),
+    WED(2),
```
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/Weekday.java]: Those values are specified in incoming data, and its "Mon, Tue..." so we can't use MONDAY, TUESDAY etc. here.

[the remark this answers is not in the document]
Comment (wojtekbu) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/Weekday.java]: Maybe renamed Weekday enum in order to specify incoming data source

File: src/main/java/org/benetech/servicenet/adapter/eden/model/Accessibility.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import com.google.gson.annotations.SerializedName;
+import lombok.Data;
+
+@Data
+public class Accessibility {
+
+    private String disabled;
+
+    @SerializedName("public")
+    private String description;
```
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/Accessibility.java]: I think it not necessary here, but I'll add it in TakeAllRequest class

File: src/main/java/org/benetech/servicenet/adapter/eden/model/BaseData.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wojtekbu) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/BaseData.java]: the explicit way seems to be more readable

File: src/main/java/org/benetech/servicenet/adapter/eden/model/Weekday.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/Weekday.java]: Its already in a separate package for this data

File: src/main/java/org/benetech/servicenet/util/HttpUtils.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+public final class HttpUtils {
+
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    public static Map<String, String> getStandardHeaders(String token) {
+        Map<String, String> headers = new HashMap<>();
+        headers.put(HttpUtils.AUTHORIZATION, HttpUtils.getBearerAuthValue(token));
```
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/util/HttpUtils.java]: No

File: src/main/java/org/benetech/servicenet/adapter/eden/model/BaseData.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/eden/model/BaseData.java]: Cool

File: src/main/java/org/benetech/servicenet/util/HttpUtils.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/util/HttpUtils.java]: I'll use org.apache.http.Header;

**Pull request merged**